magda-web-client/src/Components/Header/Header.js

```diff
@@ -17,17 +17,26 @@ class Header extends Component {
     }
 
     componentDidMount() {
+        this._isMounted = true;
         this.context.router.history.listen(() => {
-            this.setState({
-                isMobileMenuOpen: false
-            });
+            if (this._isMounted) {
```

Comment:
If we are sure the `setState` here is the one who causes the problem, we probably should solve it by un-register the listener when the component unmount (`componentWillUnmount `). Otherwise, we just muted the warning (as we will not call `setState` on an unmounted component anymore) but the memory leaks (listener) are still there.
You can de-register the listener by:
```
this.unlisten = this.context.router.history.listen(...)
```

Then call `this.unlisten` when component is unmounted.